dashi/poolqc/pooling_qc_sample.py

```diff
@@ -216,36 +301,109 @@ def update_sampleindices(run):
             'showlegend': False,
+            'barmode': 'group',
 
         }
     }
 
 
+def percent_difference(run, index_threshold):
+    data = []
+
+    for inx, d in run.groupby('library'):
+        d['Per Cent Difference'] = (d['Clusters'] - index_threshold) / index_threshold * 100
+        data.append(
+            go.Bar(
+                x=list(d['library']),
+                y=list(d['Per Cent Difference']),
+                name=inx,
+                marker={'color': '#20639B'},
+            )
+        )
+    return {
+        'data': data,
+        'layout': {
+            'title': 'Per Cent Difference of Index Clusters',
+            'xaxis': {'title': 'Sample', 'automargin': True, 'tickangle': '90', },
+            'yaxis': {'title': 'Per Cent', 'range': [-100, 100]},
+            'showlegend': False,
+        }
+    }
+
+
 @app.callback(
     [dep.Output('SampleIndices', 'figure'),
+     dep.Output('Per Cent Difference', 'figure'),
      dep.Output('Summary Table', 'columns'),
      dep.Output('Summary Table', 'data'),
-     dep.Output('Summary Table', 'style_data_conditional'),
      dep.Output('download-link', 'href'),
-     dep.Output('download-link', 'download')],
+     dep.Output('download-link', 'download'),
+     dep.Output('object_threshold', 'children'),
+     dep.Output('object_passed_samples', 'children')],
     [dep.Input('select_a_run', 'value'),
-     dep.Input('lane_select', 'value')])
-def update_graphs(run_alias, lane_alias):
+     dep.Input('lane_select', 'value'),
+     dep.Input('index_threshold', 'value'),
+     dep.Input('pass/fail', 'value'),
+     dep.Input('sample_type', 'value')])
+def update_graphs(run_alias, lane_alias, threshold, PassOrFail, sample_type):
+    """
+    Outputs:
+        SampleIndices: updates the figure Layout element of the graph titled the Index Clusters per Sample
+        Per Cent Difference: updates the figure of Layout element of Per Cent Differences of Index Clusters
+        Summary Table:
+            Columns: updates the names of columns in the Summary Table
+            Data: updates the cells in the Summary Table
+        download-link:
+            href: clickable link to download page for raw data in Summary Table
+            download: csv file for raw data in Summary Table
+        'object_threshold': updates the value of the 'Clusters Threshold' textbox at the top of report
+        'object_passed_samples': updates the value of the 'Passed Samples' textbox at the top of the report
+    Inputs:
+        :param run_alias: the value from the run drop-down in drawer
+        :param lane_alias: value selected from lane options in drawer
+        :param threshold: either the calculated initial threshold, or if changed, the new value of index_threshold
+        :param PassOrFail: calculated value of how many samples have passed dependent on the threshold
+        :param sample_type: the values of all the types of samples selected to view in the report (DNA vs RNA)
```

Comment:
Some of the Inputs don't match those of the `dep. Input`

Reply:
They match the arguments to the function. For clarity, it might make sense to set the parameters to have the same name as the Input's ID.

Reply:
You are correct. My eye was drawn to the `dep.Input` names.